api/app.go

```diff
@@ -332,7 +375,97 @@ func (app *App) Start() error {
 				zap.Int("graceperiod", graceperiod))
 		})
 		time.Sleep(time.Duration(graceperiod) * time.Millisecond)
+	case err := <-errch:
+		return err
+	case <-stopped:
 	}
 	log.I(l, "app stopped")
 	return nil
 }
+
+func (app *App) startGRPCServer(lis net.Listener, errch chan<- error) {
+	var basicAuthInterceptor grpc.UnaryServerInterceptor
+
+	basicAuthUser := app.Config.GetString("basicauth.username")
+	if basicAuthUser == "" {
+		basicAuthInterceptor = grpc.UnaryServerInterceptor(app.noAuthMiddleware)
+	} else {
+		basicAuthInterceptor = grpc_auth.UnaryServerInterceptor(app.basicAuthMiddleware)
+	}
+
+	app.grpcServer = grpc.NewServer(grpc.UnaryInterceptor(
+		grpc_middleware.ChainUnaryServer(
+			basicAuthInterceptor,
+			grpc.UnaryServerInterceptor(app.loggerMiddleware),
+			grpc.UnaryServerInterceptor(app.recoveryMiddleware),
+			grpc.UnaryServerInterceptor(app.responseTimeMetricsMiddleware),
+			grpc.UnaryServerInterceptor(app.sentryMiddleware),
+			grpc.UnaryServerInterceptor(app.newRelicMiddleware),
+		),
+	))
+	api.RegisterPodiumServer(app.grpcServer, app)
+
+	app.grpcReady <- true
+	if err := app.grpcServer.Serve(lis); err != nil {
+		errch <- fmt.Errorf("error trying to serve with grpc server: %v", err)
+	}
+}
+
+func (app *App) startHTTPServer(lis net.Listener, errch chan<- error) {
+	gatewayMux := runtime.NewServeMux(runtime.WithMarshalerOption(runtime.MIMEWildcard,
+		&runtime.JSONPb{EmitDefaults: true}))
+	opts := []grpc.DialOption{grpc.WithInsecure()}
+
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	if err := api.RegisterPodiumHandlerFromEndpoint(ctx, gatewayMux, app.GRPCEndpoint, opts); err != nil {
+		errch <- fmt.Errorf("error registering multiplexer for grpc gateway: %v", err)
```

Comment:
This looks like an error that wants to be returned promptly instead of going through the channel. Since this is more of configuration-like code, consider moving to GetApp/Configure, as suggested above.

Reply:
There's a reason for this to be place here. This function was on GetApp/Configure but there were errors appearing on the tests. The tests run GetApp/Configure long before starting the App. After I brought this function to Start the problems were gone.